modules/core/src/lib/deck.js

```diff
@@ -499,7 +499,9 @@ export default class Deck {
     this.effectManager = new EffectManager({gl, layerManager: this.layerManager});
```

Comment:
Design wise I don't think EffectManager should be aware of the LayerManager. This breaks the ownership hierarchy (Deck owns all *Managers and data flows one-way). Exposing the these sub systems to each other will also risk making the code more difficult to maintain. The EffectManager could easily offer an API such as

```js
effectManager.getModuleParameters(layer, context);
```

For `drawLayers` to call.

Reply:
ditto